plugins/plugin-svelte/plugin.js

```diff
@@ -169,6 +169,18 @@ module.exports = function plugin(snowpackConfig, pluginOptions = {}) {
 
       const compiled = svelte.compile(codeToCompile, finalCompileOptions);
       const {js, css} = compiled;
+
+      var charset = 'utf-8';
+      const mapping = (m) =>
+        '\n//# sourceMappingURL=data:application/json;charset=' +
```

Comment:
question: it feels weird to manually concatenate the sourcemap url into the file, shouldn't whatever is being used to manipulate the files and generate the sourcemaps like `magic-string`, `sourcemap`, or `recast` be the one adding this in automatically?

Reply:
I agree it does but the svelte compiler doesn't seem to expose an interface like esbuild does for defining wether you'd like the sourcemaps to be external or inlined; it just generates and outputs `{ code, map }` presumably for use further on in the build process.

So I gathered we either need to write the map to disk/cache, create a URL for it and append that to the bottom of the file, or just append the source map itself. The latter seemed much easier at the time 😅

Reply:
I not too familiar with Svelte, but it seems strange that svelte's compiler doesn't add the sourcemap url at the end of the transform for users. Svelte's creator also made rollup and magic-string, so I would have thought that the Svelte framework would be sourcemap compliant. 

Are we positive it isn't being hidden behind some Svelte compiler api flag? Have we tried using `outputFilename` compiler option and seeing if it appends the sourcemap url for us? https://svelte.dev/docs#svelte_compile

Reply:
Looks like the esbuild svelte plugin also appends the url 😞  https://esbuild.github.io/plugins/#svelte-plugin

Reply:
Thanks for looking into this 🙇 very reassuring to know we are not alone in doing this appendage (I admit it did feel like a bit of a smell). Perhaps we make this a util function, something like `inlineSourceMap(code, map)`.